src/test/java/school/lemon/changerequest/java/junit/hw1/task1/api/ContactApiTest.java

```diff
@@ -0,0 +1,215 @@
+package school.lemon.changerequest.java.junit.hw1.task1.api;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import school.lemon.changerequest.java.junit.hw1.task1.model.*;
+
+import static org.apache.commons.lang3.RandomStringUtils.randomNumeric;
+import static org.junit.Assert.*;
+
+public class ContactApiTest {
+
+    ContactApi contactApi = null;
+    Contact testContact = null;
+
+    @Before
+    public void setUp() {
+        contactApi = new ContactApiImpl(new ContactIdGeneratorImpl());
```

Comment:
What about having `mock` of `ContactIdGenerator` instead of the real implementation?

Reply:
Can you help me?
How can mock <code>thenReturn</code> different values? Construction like
<code>ContactIdGenerator generatorID = Mockito.spy(ContactIdGeneratorImpl.class);
        Mockito.when(generatorID.generate()).thenReturn(i++);</code>
doesn`t work.

Reply:
1. Do you really need `Spy` of the `ContactIdGenerator`?

Spies are used whenever you want to partially mock the behavior. In your case you can mock the entire object and use `Mockito.mock()` instead of `Mockito.spy()`.

2. Mockito has different syntax for stubbing methods on `spy` instances:

For your particular example it'll be smth like:
```java
Mockito.doReturn(i++).when(generatorID).generate()
```

Also, your variable name is confusing. I'd better name it `idGenerator`.